master/internal/trial.go

```diff
@@ -910,6 +978,12 @@ func (t *trial) resetTrial(
 			"ignoring trial runner failure since termination was requested",
 		)
 		return
+	case t.cancelUnready:
```

Comment:
non-blocking: maybe rename this to `terminatingBeforeConnecting`

Reply:
This sounds a bit confusing in that connecting isn't a trial's behavior. We lack such a state that describes something between "assigned" and "training". "ready" sounds a bit more accurate.

Reply:
Yeah that sounds right, don't love `terminatingBeforeConnecting` so fine staying with `cancelUnready`.